src/kubernetes_cluster/spec/builtin_controllers/garbage_collector.rs

```diff
@@ -23,6 +23,20 @@ verus! {
 
 impl <K: ResourceView, E: ExternalAPI, R: Reconciler<K, E>> Cluster<K, E, R> {
 
+pub open spec fn garbage_collector_deletion_enabled(key: ObjectRef) -> StatePred<Self> {
```

Comment:
Why not just use it in the `transition` below?

Reply:
Actually we can even make it the `precondition` and the `transition` is as simple as sending a delete request

Reply:
I am not sure whether this will change some place in the proof. Let's leave it for later.